spring-batch-core/src/main/java/org/springframework/batch/core/step/item/SimpleChunkProcessor.java

```diff
@@ -292,16 +304,32 @@ protected void write(StepContribution contribution, Chunk<I> inputs, Chunk<O> ou
 			 * here, so prevent any more processing of these inputs.
 			 */
 			inputs.clear();
+			status = STATUS_FAILURE;
 			throw e;
 		}
+		finally {
+			stopTimer(sample, contribution.getStepExecution(), "chunk.write", status, "Chunk writing");
+		}
 		contribution.incrementWriteCount(outputs.size());
 	}
 
+	private void stopTimer(Timer.Sample sample, StepExecution stepExecution, String metricName, String status, String description) {
+		sample.stop(Timer.builder(METRICS_PREFIX + metricName)
+				.description(description + " duration in seconds")
+				.tag("job.name", stepExecution.getJobExecution().getJobInstance().getJobName())
+				.tag("step.name", stepExecution.getStepName())
+				.tag("status", status)
+				.register(Metrics.globalRegistry)
+		);
+	}
+
 	protected Chunk<O> transform(StepContribution contribution, Chunk<I> inputs) throws Exception {
 		Chunk<O> outputs = new Chunk<>();
 		for (Chunk<I>.ChunkIterator iterator = inputs.iterator(); iterator.hasNext();) {
 			final I item = iterator.next();
 			O output;
+			Timer.Sample sample = Timer.start(Metrics.globalRegistry);
```

Comment:
@jkschneider Should this be moved outside the loop? Do we need a sample for each processed item?

Reply:
I suppose that's up to you and what you want to monitor. No right/wrong answer here.